Index: typo3/sysext/tstemplate/ts/index.php
===================================================================
--- typo3/sysext/tstemplate/ts/index.php	(revision 2717)
+++ typo3/sysext/tstemplate/ts/index.php	(working copy)
@@ -117,6 +117,14 @@
 		</script>
 		';
 
+			// include t3editor if its loaded
+		if (t3lib_extMgm::isLoaded('t3editor')) {
+			$GLOBALS['T3_VAR']['t3editorObj'] = t3lib_div::getUserObj('EXT:t3editor/class.tx_t3editor.php:&tx_t3editor');
+			if (is_object($GLOBALS['T3_VAR']['t3editorObj'])) {
+				$this->doc->JScode.= $GLOBALS['T3_VAR']['t3editorObj']->getJavascriptCode();
+			}
+		}		
+
 			$this->doc->postCode='
 		<script language="javascript" type="text/javascript">
 			script_ended = 1;
Index: typo3/sysext/tstemplate_info/class.tx_tstemplateinfo.php
===================================================================
--- typo3/sysext/tstemplate_info/class.tx_tstemplateinfo.php	(revision 2717)
+++ typo3/sysext/tstemplate_info/class.tx_tstemplateinfo.php	(working copy)
@@ -31,6 +31,9 @@
 require_once(PATH_t3lib."class.t3lib_extobjbase.php");
 
 class tx_tstemplateinfo extends t3lib_extobjbase {
+
+	var $tce_processed = false;  // indicator for t3editor, if data is stored
+	
 	function modMenu()	{
 		global $LANG;
 
@@ -234,6 +237,9 @@
 						// Clear the cache (note: currently only admin-users can clear the cache in tce_main.php)
 					$tce->clear_cacheCmd("all");
 
+						// tce were processed successfully
+					$this->tce_processed = true;
+
 						// re-read the template ...
 					$this->initialize_editor($this->pObj->id,$template_uid);
 				}
@@ -264,7 +270,28 @@
 					}
 				}
 			}
+			
+		// response ajax call by t3editor
+            if (is_object($GLOBALS['T3_VAR']['t3editorObj'])) {
+                    // if TSEditor is enabled
+                	// this POST is a Ajax-Request
+            	if ($GLOBALS['T3_VAR']['t3editorObj']->isEnabled
+            		&& t3lib_div::_POST('submitAjax')) {
 
+			$GLOBALS['T3_VAR']['t3editorObj']->setBEUCdisableT3Editor(false);
+
+            	       		// abort request here and return message
+				// TODO JSON or other response here!?
+                	if ($this->tce_processed) {
+            			echo "OK";
+                	} else {
+                		echo "ERROR";
+                	}
+            		exit();
+            	}
+            }
+			
+
 			$theOutput.=$this->pObj->doc->spacer(5);
 			$theOutput.=$this->pObj->doc->section("Template information:",'<img src="'.$BACK_PATH.t3lib_iconWorks::getIcon("sys_template",$tplRow).'" width=18 height=16 align=top><b>'.htmlspecialchars($tplRow["title"]).'</b>'.htmlspecialchars(trim($tplRow["sitetitle"])?' - ('.$tplRow["sitetitle"].')':''),0,1);
 			if ($manyTemplatesMenu)	{
@@ -368,8 +395,19 @@
 				}
 			}
 			if ($e["config"])	{
-				$outCode='<textarea name="data[config]" rows="'.$numberOfRows.'" wrap="off" class="fixed-font enable-tab"'.$this->pObj->doc->formWidthText(48,"width:98%;height:70%","off").' class="fixed-font">'.t3lib_div::formatForTextarea($tplRow["config"]).'</textarea>';
-
+				if (is_object($GLOBALS['T3_VAR']['t3editorObj'])
+					&& $GLOBALS['T3_VAR']['t3editorObj']->isEnabled) {
+					$outCode=$GLOBALS['T3_VAR']['t3editorObj']->getCodeEditor(
+								'data[config]',	// name
+								'fixed-font',	// class
+								t3lib_div::formatForTextarea($tplRow["config"]),	// content
+								'rows="'.$numberOfRows.'" wrap="off" '.$this->pObj->doc->formWidthText(48,"width:98%;height:60%","off"),
+								'Template: '.htmlspecialchars($tplRow["title"]).': Setup'
+								);
+				} else {
+					$outCode='<textarea name="data[config]" rows="'.$numberOfRows.'" wrap="off" class="fixed-font enable-tab"'.$this->pObj->doc->formWidthText(48,"width:98%;height:70%","off").' class="fixed-font">'.t3lib_div::formatForTextarea($tplRow["config"]).'</textarea>';
+				}
+				
 				if (t3lib_extMgm::isLoaded("tsconfig_help"))	{
 					$url=$BACK_PATH."wizard_tsconfig.php?mode=tsref";
 					$params=array();
